### ISSUE TITLE
bug: epubs with percent encoded files in the manifest are not parsed correctly

### ISSUE THREAD

**Issue by morbatex** [danigm/epub-rs#9]
Hi, 

I think I found a bug in when reading epubs where files in the manifest use percent encoded filenames or paths, while the matching files and directories in the archieve are "normal".

For example if I try to read an epub containing the file `a file (with % encoding).html` with and the following manifest

```
<?xml version="1.0" encoding="utf-8" standalone="yes"?>
<package xmlns="http://www.idpf.org/2007/opf" unique-identifier="bookid" version="2.0" xmlns:dc="http://purl.org/dc/elements/1.1/">
...
  <manifest>
    <item href="a%20file%20%28with%20%25%20encoding%29.html" id="SomeId1" media-type="application/xhtml+xml" />
    ...
  </manifest>
...
</package>
```
I cannot access the corresponding page but get a zip::result::ZipError::FileNotFound error.

If epub-rs tries to percent decode a path when a FileNotFound error occured it could read more files.

**Comment by danigm** [comment]
Yes, it's a bug, I resolved the exact same problem in [libgepub](https://gitlab.gnome.org/GNOME/libgepub/issues/7#note_366794)

The easier way to solve this is to unescape the string.

**Comment by morbatex** [comment]
If the string is unescaped before the lib tries to access the it could not access a file named `a%20normal%20file.xml` in the archieve because the `%20` will get unescaped into a space. I have some ebooks, which contain some files with percent encoding in the name.
